src/ucp/core/ucp_mm.c

```diff
@@ -1463,30 +1463,22 @@ void ucp_mem_rcache_cleanup(ucp_context_h context)
     }
 }
 
-ucs_status_t
-ucp_mm_get_alloc_md_index(ucp_context_h context, ucp_md_index_t *md_idx)
+ucs_status_t ucp_mem_get_host_alloc_md_index(ucp_context_h context)
 {
-    const ucs_memory_type_t alloc_mem_type = UCS_MEMORY_TYPE_HOST;
     unsigned memh_reg_flags;
     ucp_mem_h alloc_memh;
     ucs_status_t status;
 
-    if (!context->alloc_md_index_initialized) {
-        memh_reg_flags = UCT_MD_MEM_ACCESS_RMA | UCT_MD_MEM_FLAG_HIDE_ERRORS;
-        status         = ucp_memh_alloc(context, NULL, 1, alloc_mem_type,
-                                        memh_reg_flags, "get_alloc_md_id",
-                                        &alloc_memh);
-        if (status != UCS_OK) {
-            return status;
-        }
-
-        context->alloc_md_index_initialized     = 1;
-        context->alloc_md_index[alloc_mem_type] = alloc_memh->alloc_md_index;
+    memh_reg_flags = UCT_MD_MEM_ACCESS_RMA | UCT_MD_MEM_FLAG_HIDE_ERRORS;
+    status         = ucp_memh_alloc(context, NULL, 1, UCS_MEMORY_TYPE_HOST,
```

Comment:
this may trigger set (or using) GPU device context during ucp_init() and we don't want that
maybe we can find the alloc md index without doing cudaHostRegister? or even without actually allocating anything?

Reply:
Updated without the registration part, but still exercising alloc path. cuda/rocm alloc early refuse host memory allocation so we should be good on that side? But if we want for instance to be able to use sysv/huge tlb actual function, I did not see simple way to test without allocating. Else we could just pick mmap, no alloc, but I think this would depart from current behavior.